src/main/scala/com/gu/fezziwig/CirceScroogeMacros.scala

```diff
@@ -165,17 +219,39 @@ class CirceScroogeMacrosImpl(val c: blackbox.Context) {
 
       val implicitDecoderForParam: c.Tree = getImplicitDecoder(paramType)
 
-      cq"""$paramName => c.downField($paramName).success.flatMap(_.as[$paramType]($implicitDecoderForParam).toOption).map($applyMethod)"""
+      val decExpr = {
+        cq"""$paramName =>
```

Comment:
See 'case clause' in http://docs.scala-lang.org/overviews/quasiquotes/syntax-summary.html